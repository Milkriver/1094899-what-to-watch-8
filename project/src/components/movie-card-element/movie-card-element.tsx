import { useHistory } from 'react-router';
import { AppRoute } from '../../const';
import VideoPlayer from '../videoplayer/videoplayer';

interface IProps {
  filmName: string
  id: number
  videoLink: string
  onMouseOver: (id: number | undefined) => void
  posterSrc: string
  activeCardId: number | undefined
}

export function MovieCardElement({ activeCardId, onMouseOver, filmName, posterSrc, id, videoLink }: IProps): JSX.Element {
  const history = useHistory();
  const onClick = () => {
    if (activeCardId === undefined) {
      return;
    }
    const url = AppRoute.Film.replace(':id', activeCardId.toString());
    history.push(url);
  };
  return (
    <article className="small-film-card catalog__films-card" onClick={onClick} onMouseEnter={() => onMouseOver(id)} onMouseOut={() => onMouseOver(undefined)}>
      <div className="small-film-card__image" >
        {activeCardId === id ? <VideoPlayer posterSrc={posterSrc} videoLink={videoLink} /> :
          <img src={posterSrc} alt='' width="280" height="175" />};
      </div>
      <h3 className="small-film-card__title">
        <div className="small-film-card__link">{filmName}</div>
      </h3>
    </article>
  );
}

