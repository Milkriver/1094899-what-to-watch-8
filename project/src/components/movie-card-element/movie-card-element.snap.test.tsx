import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MovieCardElement } from './movie-card-element';
import { configureMockStore } from '@jedmao/redux-mock-store';

const mockFn = () => Promise.resolve();
describe('Component: MovieCardElement', () => {
  const mockStore = configureMockStore();

  const store = mockStore({
    MOVIES: {},
    USER: {},
    REVIEWS: {},
  });

  store.dispatch = jest.fn();

  test('should render correctly', () => {
    window.HTMLMediaElement.prototype.play = jest.fn();
    const page = (
      <Provider store={store}>
        <MovieCardElement activeCardId={1} onMouseOver={mockFn} filmName={'movieName'} posterSrc={'movieUrl'} id={2} videoLink={'videoLink'} />
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
