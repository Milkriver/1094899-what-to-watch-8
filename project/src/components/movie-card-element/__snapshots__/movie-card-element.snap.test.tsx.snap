// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MovieCardElement should render correctly 1`] = `
<div>
  <article
    class="small-film-card catalog__films-card"
  >
    <div
      class="small-film-card__image"
    >
      <img
        alt=""
        height="175"
        src="movieUrl"
        width="280"
      />
      ;
    </div>
    <h3
      class="small-film-card__title"
    >
      <div
        class="small-film-card__link"
      >
        movieName
      </div>
    </h3>
  </article>
</div>
`;
